client/src/components/Invites.js

```diff
@@ -46,24 +57,126 @@ const Invites = (props) => {
     setShowRequests(false);
   };
 
+  //both of these will need to make API calls to save modified state
+  const handleApproveRequest = async (invite) => {
+    try {
+      const res = await axios.put(`/invitation/${invite._id}/approve`);
+      return res;
+    } catch (err) {
+      console.log(err.message);
+    }
+  };
+  const handleRejectRequest = async (invite) => {
+    try {
+      const res = await axios.put(`/invitation/${invite._id}/reject`);
+      return res;
+    } catch (err) {
+      console.log(err.message);
+    }
+  };
+  useEffect(() => {
+    //axios fetch all the PENDING invites user has sent and received
+    //is called whenever a change in invitations takes place
+    (async function fetchPendingInvites() {
+      try {
+        const res = await axios.get(
+          `user/${userState.user.id}/invitations/pending`,
+        );
+
+        //if response is ok or user has no invites
+        if (res.status === 200 || res.status === 204) {
+          //sets invites to an object of pending in and out invites
+          const pendingInvites = res.data;
+          return setInvites(pendingInvites);
+        }
+        if (res.status === 404 || res.status === 500) {
+          //user not found
+          //dispatch user error
+          setInvitesErrorMsg('Error fetching invites');
+          setInvites(true);
+        }
+      } catch (err) {
+        console.log(err.message);
+        setInvitesErrorMsg('Error fetching invites');
+        setInvites(true);
+        //server error
+        //dispatch user error
+      }
+    })();
+  }, [invites, userState.user.id]);
```

Comment:
Do we need to do this every time invites changes, or just if the user changes?

Reply:
At this point, I was just following the console warnings but in my opinion, we only need this to change when invites change.

Reply:
I don't know that we need to make an API call every time the invites change. What if we made this call once when the page loaded and then just kept track locally on the front end any changes made by accepting or rejecting invites?

Reply:
Do you mean something locally like splicing the arrays to update them?

Reply:
yup, exactly